plugins/output-redis/src/main/scala/com/stratio/sparkta/plugin/output/redis/RedisOutput.scala

```diff
@@ -57,20 +54,31 @@ class RedisOutput(keyName: String,
    * @param metricOperations that will be saved.
    */
   override def upsert(metricOperations: Iterator[(DimensionValuesTime, Map[String, Option[Any]])]): Unit = {
-    metricOperations.toList.groupBy(upMetricOp =>
-      AggregateOperations.keyString(upMetricOp._1, upMetricOp._1.timeDimension, fixedDimensions))
-      .filter(_._1.size > 0).foreach(collMetricOp => {
-      collMetricOp._2.foreach(metricOp => {
-        val timeDimension = metricOp._1.timeDimension
-        val hashKey = collMetricOp._1 + IdSeparator + metricOp._1.dimensionValues
-          .map(dimVal => List(dimVal.getNameDimension, dimVal.value.toString))
-          .flatMap(_.toSeq).mkString(IdSeparator) + IdSeparator + timeDimension + IdSeparator + metricOp._1.time
+    metricOperations.toList.groupBy { case (dimensionsTime, aggregations) =>
```

Comment:
IMHO, a refactor is a good opportunity to make this more readable and maintainable, don't you think?

map().filter().foreach( case { ... } ) Too convoluted